### DISCUSSION TITLE
`fullWsTransport` and mutations result in `MER_ERR_METHOD_NOT_ALLOWED`?

### DISCUSSION THREAD

**Discussion by tomi-bigpi** [mercurius-js/mercurius#977] [Q&A]
Based on the documentation for `fullWsTransport` it seems like you should be able to perform mutations over web sockets, but in practice this results in a "Method not allowed" error.

This is happening because of the check here: https://github.com/mercurius-js/mercurius/blob/master/index.js#L474

It checks whether the raw request is `GET` and doesn't allow mutations. But the raw request in the case of web sockets is the original GET request that receives a 101 Switching protocols / connection upgrade.

Is this intended behavior?

As a side note, this actually results in an invalid response that at least Apollo client doesn't like and errors with `ApolloError: "error" message expects the 'payload' property to be an array of GraphQL errors, but got "Method not allowed"`

**Comment by mcollina** [comment]
Good spot! Would you like to send a PR to fix?

**Reply by tomi-bigpi** [reply]
PR here: https://github.com/mercurius-js/mercurius/pull/986
